src/dex/aave-v1/aave-v1.ts

```diff
@@ -0,0 +1,217 @@
+import {
+  Token,
+  Address,
+  ExchangePrices,
+  AdapterExchangeParam,
+  SimpleExchangeParam,
+  PoolLiquidity,
+  Logger,
+} from '../../types';
+import { SwapSide, Network } from '../../constants';
+import { getDexKeysWithNetwork } from '../../utils';
+import { IDex } from '../../dex/idex';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import { AaveV1Data, AaveV1Param } from './types';
+import { SimpleExchange } from '../simple-exchange';
+import { AaveV1Config, Adapters } from './config';
+import AAVE_LENDING_POOL_ABI_V1 from '../../abi/AaveV1_lending_pool.json';
+import ERC20 from '../../abi/erc20.json';
+import { isAAVEPair } from './tokens';
+
+const AaveGasCost = 400 * 1000;
+
+enum AaveV1Functions {
+  deposit = 'deposit',
+  redeem = 'redeem',
+}
+
+const AAVE_LENDING_POOL = '0x398eC7346DcD622eDc5ae82352F02bE94C62d119';
+const AAVE_PROXY = '0x3dfd23a6c5e8bbcfc9581d2e864a68feb6a076d3';
+const REF_CODE = 1;
+
+export class AaveV1
+  extends SimpleExchange
+  implements IDex<AaveV1Data, AaveV1Param>
+{
+  readonly hasConstantPriceLargeAmounts = true;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(AaveV1Config);
+
+  logger: Logger;
+  aavePool: Interface;
+  aContract: Interface;
+
+  constructor(
+    protected network: Network,
+    protected dexKey: string,
+    protected dexHelper: IDexHelper, // TODO: add any additional optional params to support other fork DEXes
+  ) {
+    super(dexHelper.augustusAddress, dexHelper.provider);
+    this.logger = dexHelper.getLogger(dexKey);
+    this.aavePool = new Interface(AAVE_LENDING_POOL_ABI_V1 as JsonFragment[]);
+    this.aContract = new Interface(ERC20 as JsonFragment[]);
+  }
+
+  // Initialize pricing is called once in the start of
+  // pricing service. It is intended to setup the integration
+  // for pricing requests. It is optional for a DEX to
+  // implement this function
+  async initializePricing(blockNumber: number) {}
+
+  // Returns the list of contract adapters (name and index)
+  // for a buy/sell. Return null if there are no adapters.
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return Adapters[this.network][side];
+  }
+
+  // Returns list of pool identifiers that can be used
+  // for a given swap. poolIdentifers must be unique
+  // across DEXes. It is recommended to use
+  // ${dexKey}_${poolAddress} as a poolIdentifier
+  async getPoolIdentifiers(
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    blockNumber: number,
+  ): Promise<string[]> {
+    const aToken = isAAVEPair(this.network, srcToken, destToken);
+    if (aToken === null) {
+      return [];
+    }
+    const tokenAddress = [
+      srcToken.address.toLowerCase(),
+      destToken.address.toLowerCase(),
+    ]
+      .sort((a, b) => (a > b ? 1 : -1))
+      .join('_');
+
+    const poolIdentifier = `${this.dexKey}_${tokenAddress}`;
+    return [poolIdentifier];
```

Comment:
Though this would logically work it would confuse the DEVs. The pool identifier should be simply the `${this.dexKey}_${aTokenAddress}` as suggested by @ColonelJ  before